### PR TITLE
Use existing context

### PR DESCRIPTION
**Pull request by hgw77**
* I have the use case that we do the authentication outside of misty so we do not need a second auth with misty
* bypass auth in misty if existing context is available. With context I mean a valid token, catalog and expire date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,37 +125,41 @@ The URL and credentials details are necessary to authenticate with the identity
...
+* `:user_domain`
+  User domain name
 * `:password`
   Password for user. Cannot be used together with `:token`.
 * `:token`
   Previous Keystone v3 token for user. Can only be used with Keystone v3. Overrides all user and password parameters.
+* `:context`
+  I you have a proper context with `token`, `catalog` and `expire date` you can bypass the authentication to save the api call
+  Overrides all user and password parameters
+  Example: ``{:context => { :token => token, :catalog => catalog, :expires_at => expires_at }}``
```
Comment (Carthaca) [inline README.md]: I guess you mean `expires`, should match https://github.com/sapcc/misty/blob/fa1d200173b1931602a52f752ed502ceb1744380/lib/misty/auth.rb#L38

**Comment by gildub** [comment]
If there is no authentication needed since token, catalog and expiry are provided then there is not real need for triggering Auth object. What about an approach to provide those to misty as object (@auth)? 

We also need to validate those provided datum, what happens if the catalog is faulty?
 
It looks like your editor automatically replaces trailing spaces which is fine on most files but not on markup files like README.md, where double spaces means a new line for md markup.

**Comment by hgw77** [comment]
sorry for the delay! 

about the auth object, do you mean to create a misty auth object first like 


    auth = Misty::AuthV3.new( context: { 
      catalog: current_user.context["catalog"],
      expires: current_user.context["expires_at"],
      token: current_user.token
    })

and provide it to ``Misty::Cloud.new()``? If yes whats the advantage instead of provide the token, catalog and expire date directly?

if the catalog is faulty I guess it crash but I never saw that because in my case the catalog is generated from the keystone response

**Comment by gildub** [comment]
@hgw77,

Yes I was thinking about an object but that doesn't effectively make sense from the parameter viewpoint.

Thanks for fixing the spaces.

**Comment by hgw77** [comment]
👍

**Pull request merged**